### PR TITLE
Add `hypre_ParCSRMatrixBlkFilterDevice`

### PR DESCRIPTION
**Pull request by victorapm**
Add device implementation for `hypre_ParCSRMatrixBlkFilter`. See #1114

A few results obtained with one V100 (lassen):

pr1119_setup
pr1119_solve

### CODE REVIEW

**Review by rfalgout — APPROVED** [review]
Looks good.  Thanks!

**Review by oseikuffuor1 — APPROVED** [review]
Looks good to me. Thanks.

**Comment by victorapm** [comment]
@liruipeng could you take a look at this? Thanks!

File: src/parcsr_mv/par_csr_filter_device.c [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+         hypre_mask         ballot     = hypre_ballot_sync(HYPRE_WARP_FULL_MASK, write);
+         HYPRE_Int          laneoff    = hypre_popc(ballot & ((hypre_mask_one << lane) - 1));
+
+         if (write)
+         {
+            HYPRE_Int idx = offd_offset + laneoff;
+            B_offd_j[idx] = col;
+            B_offd_a[idx] = A_offd_a[j];
+
+#ifndef HYPRE_USING_SYCL
```
Comment (liruipeng) [inline src/parcsr_mv/par_csr_filter_device.c:174]: a minor thing: maybe `#if !defined`. Is it an SSDK requirement?

**Review by liruipeng — APPROVED** [review]
Looks good other than a minor comment. Thanks!

File: src/parcsr_mv/par_csr_filter_device.c (hunk 1 above)
Comment (victorapm) [reply src/parcsr_mv/par_csr_filter_device.c:174]: It doesn't seem to be an xSDK requirement, but I can make the change

Comment (liruipeng) [reply src/parcsr_mv/par_csr_filter_device.c:174]: If not, no need to change. but I do remember we talked about `#if defined()` is preferred over`#ifdef` by some software rule. I don't know why though.

**Pull request merged**